lib/phoenix_swagger/schema_definition.ex

```diff
@@ -0,0 +1,85 @@
+defmodule PhoenixSwagger.SchemaDefinition do
+  @moduledoc """
+  Provides a DSL to define Swagger Schemas outside of `swagger_definitions/1`. This separates the definition of a schema
+  from the placement inside of swagger.
+
+  # Example:
+
+  defmodule Example.Schemas.Book do
+    use PhoenixSwagger.SchemaDefinition
+
+    swagger_schema_definition do
+      title :string, "Title", required: true
+      author :string, "Author", required: true
+    end
+  end
+  """
+  alias PhoenixSwagger.Schema
+
+  defmacro __using__(_opts) do
+    quote do
+      import PhoenixSwagger.SchemaDefinition
+    end
+  end
+
+  @doc """
+  This macro accepts a block where the schema is defined by listing the properties using a DSL.
+  Defining a property for the schema takes 2 required parameters and one optional parameter.
+
+  The first parameter is the name of the property.
+  The second parameter is the property type.
+  The third optional parameter is an optional list of parameters accepting values for `required`, `enum`, and `format`.
```

Comment:
This doc looks a little out of date.
The description string appears to be the required 3rd parameter, with other options in the 4th optional parameter?